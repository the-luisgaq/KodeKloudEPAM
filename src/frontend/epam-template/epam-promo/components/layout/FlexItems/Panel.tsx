import * as uui from '@epam/uui';
import { withMods, VPanelProps } from '@epam/uui-core';

export interface PanelMods {
    shadow?: boolean;
    margin?: '24';
    background?: 'white' | 'gray90' | uui.PanelProps['background'];
}

export interface PanelProps extends VPanelProps, PanelMods {}

export const Panel = withMods<VPanelProps, PanelProps>(uui.Panel, (props) => [props.background && `uui-color-${props.background}`]);
