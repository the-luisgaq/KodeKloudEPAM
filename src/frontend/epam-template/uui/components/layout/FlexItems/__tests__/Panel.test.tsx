import React from 'react';
import { Panel } from '../Panel';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';

describe('Panel', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<Panel />);
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', async () => {
        const tree = await renderSnapshotWithContextAsync(<Panel onClick={ () => {} } margin="24" shadow />);
        expect(tree).toMatchSnapshot();
    });
});
