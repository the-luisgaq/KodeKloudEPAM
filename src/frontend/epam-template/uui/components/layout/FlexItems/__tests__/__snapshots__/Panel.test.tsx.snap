// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="uui-panel root container"
  />
</DocumentFragment>
`;

exports[`Panel should be rendered correctly with props 1`] = `
<DocumentFragment>
  <div
    class="uui-panel root shadow margin-24 container -clickable"
  />
</DocumentFragment>
`;
